### PR TITLE
br: update version (#3906)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #3906 to release-3.1

---

Signed-off-by: Neil Shen <overvenus@gmail.com>

### What is changed, added or deleted? (Required)

Update TiDB version in BR page.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: N/A
- Other reference link(s): N/A

### CODE REVIEW

File: br/backup-and-restore-use-cases.md [hunk 1]
```diff
@@ -27,13 +27,18 @@ aliases: ['/docs-cn/v3.1/reference/tools/br/use-cases/','/docs-cn/v3.1/how-to/ma
 
 ### 部署方式
 
-推荐使用 [TiDB Ansible](/online-deployment-using-ansible.md) 部署 TiDB 集群，再下载 [TiDB Toolkit](/download-ecosystem-tools.md#快速备份和恢复br) 获取 BR 应用。
+推荐使用 [TiUP](/tiup/tiup-cluster.md) 部署 TiDB 集群，再下载 [TiDB Toolkit](/download-ecosystem-tools.md#快速备份和恢复br) 获取 BR 应用。
```
Comment (TomShawn) [inline br/backup-and-restore-use-cases.md:30]: @overvenus 在 3.1 分支下是否需要调整下描述？

File: br/backup-and-restore-use-cases.md [hunk 2]
```diff
@@ -27,13 +27,18 @@ aliases: ['/docs-cn/v3.1/reference/tools/br/use-cases/','/docs-cn/v3.1/how-to/ma
...
-* PD: v3.1.0-beta.1
-* br: v3.1.0-beta.1
+* TiDB: v4.0.2
+* TiKV: v4.0.2
+* PD: v4.0.2
+* BR: v4.0.2
+
+> **注意：**
+>
+> v4.0.2 为编写本文档时的最新版本。推荐读者使用[最新版本 TiDB/TiKV/PD/BR](/releases/release-notes.md)，同时需要确保 BR 版本和 TiDB **相同**。
```
Comment (overvenus) [inline br/backup-and-restore-use-cases.md:41]:
```suggestion
> v3.1.2 为编写本文档时的最新版本。推荐读者使用[最新版本 TiDB/TiKV/PD/BR](/releases/release-notes.md)，同时需要确保 BR 版本和 TiDB **相同**。
```

File: br/backup-and-restore-use-cases.md [hunk 3]
```diff
@@ -27,13 +27,18 @@ aliases: ['/docs-cn/v3.1/reference/tools/br/use-cases/','/docs-cn/v3.1/how-to/ma
...
 
 ### 集群版本
 
-* TiKV: v3.1.0-beta.1
-* PD: v3.1.0-beta.1
-* br: v3.1.0-beta.1
+* TiDB: v4.0.2
+* TiKV: v4.0.2
+* PD: v4.0.2
+* BR: v4.0.2
```
Comment (overvenus) [inline br/backup-and-restore-use-cases.md:37]:
```suggestion
* TiDB: v3.1.2
* TiKV: v3.1.2
* PD: v3.1.2
* BR: v3.1.2
```

File: br/backup-and-restore-use-cases.md (hunk 1 above)
Comment (overvenus) [reply br/backup-and-restore-use-cases.md:30]: 需要。
```suggestion
推荐使用 [TiDB Ansible](/online-deployment-using-ansible.md) 部署 TiDB 集群，再下载 [TiDB Toolkit](/download-ecosystem-tools.md#快速备份和恢复br) 获取 BR 应用。
```

**Pull request merged**